src/components/panel/panel.js

```diff
@@ -1563,22 +1624,161 @@ MdPanelAnimation.prototype.closeTo = function(closeTo) {
+        transitionOutClass: '_md-panel-animate-slide-out'
+      };
+      reverseAnimationOptions = {
+        transitionOutClass: '_md-panel-animate-slide-in _md-panel-shown',
+        transitionInClass: '_md-panel-animate-slide-out'
+      };
+      openFrom = animator.toTransformCss(animator.calculateSlideToOrigin(
+          panelEl, this._openFrom) || "");
+      closeTo = animator.toTransformCss(animator.calculateSlideToOrigin(
+          panelEl, this._closeTo));
+      break;
+    case 'md-panel-animate-scale':
+      animationOptions = {
+        transitionInClass: '_md-panel-animate-scale-in _md-panel-shown',
+        transitionOutClass: '_md-panel-animate-scale-out'
+      };
+      reverseAnimationOptions = {
+        transitionOutClass: '_md-panel-animate-scale-in _md-panel-shown',
+        transitionInClass: '_md-panel-animate-scale-out'
+      };
+      openFrom = animator.toTransformCss(animator.calculateZoomToOrigin(
+          panelEl, this._openFrom) || "");
+      closeTo = animator.toTransformCss(animator.calculateZoomToOrigin(
+          panelEl, this._closeTo));
+      break;
+    case 'md-panel-animate-fade':
+      animationOptions = {
+        transitionInClass: '_md-panel-animate-fade-in _md-panel-shown',
+        transitionOutClass: '_md-panel-animate-fade-out'
+      };
+      reverseAnimationOptions = {
+        transitionOutClass: '_md-panel-animate-fade-in _md-panel-shown',
+        transitionInClass: '_md-panel-animate-fade-out'
+      };
+      break;
+    default:
+      if (angular.isString(this._animationClass)) {
+        animationOptions = {
+          transitionInClass: this._animationClass + ' _md-panel-shown'
+        };
+      } else {
+        animationOptions = {
+          transitionInClass: this._animationClass['open'] + ' _md-panel-shown'
+        };
+        reverseAnimationOptions = {
+          transitionInClass: this._animationClass['close']
+        };
+      }
+  }
+
+  var self = this;
+  return animator
+      .translate3d(panelEl, openFrom, openTo, animationOptions)
+      .then(function () {
+        self._reverseAnimation = function () {
+          return animator
+              .translate3d(panelEl, closeFrom, closeTo,
+                  reverseAnimationOptions);
+
+        };
+      });
+};
+
+
+/**
+ * Animate the panel close.
+ * @param $q
+ * @returns {!angular.$q.Promise}
+ */
+MdPanelAnimation.prototype.animateClose = function($q) {
+  if (this._reverseAnimation) {
+    return this._reverseAnimation();
+  }
+  return $q.reject('No panel close animation. ' +
+      'Have you called MdPanelAnimation.animateOpen()?');
+};
+
+
+/**
+ * Set the height and width to match the panel if not provided.
+ * @param {!angular.JQLite} panelEl
+ * @private
+ */
+MdPanelAnimation.prototype._fixBounds = function(panelEl) {
+  var panelWidth = panelEl[0].offsetWidth;
+  var panelHeight = panelEl[0].offsetHeight;
+
+  if (angular.isUndefined(this._openFrom.bounds.height)) {
```

Comment:
Maybe just use `== null`, which is shorter and captures both `null` and `undefined`?